src/main/scala-2.11/markets/clearing/engines/CDAMatchingEngine.scala

```diff
@@ -114,18 +113,17 @@ class CDAMatchingEngine(askOrdering: PriceOrdering[AskOrder],
         val residualQuantity = incoming.quantity - bidOrder.quantity
         val price = formPrice(incoming, bidOrder)
 
-        if (residualQuantity < 0) {
-          val totalMatch = TotalMatch(bidOrder, incoming, price)
-          // add residualOrder back into orderBook!
-          val residualOrder = bidOrder.split(-residualQuantity)
+        if (residualQuantity < 0) { // incoming order is smaller than existing order!
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1539841661/issues/source?bid=2921361&fileBranchId=3014278#l116)